pkg/module/build.go

```diff
@@ -79,16 +81,27 @@ func CreateArchive(fs vfs.FileSystem, path string, def *Definition) (*comparch.C
 }
 
 func addSources(ctx cpi.Context, cd *ocm.ComponentDescriptor, def *Definition) error {
-	src, err := git.Source(ctx, def.Source, def.Repo, def.Version)
-	if err != nil {
-		return err
-	}
-
-	if idx := cd.GetSourceIndex(&src.SourceMeta); idx < 0 {
-		cd.Sources = append(cd.Sources, *src)
-	} else {
-		cd.Sources[idx] = *src
+	if gitPath, ok, _ := files.FindDirectoryContaining(def.Source, ".git"); ok {
+		gitSource := source.NewGitSource()
+		src, err := fetchSource(ctx, gitSource, gitPath, def.Repo, def.Version)
+
+		if err != nil {
+			return err
+		}
+		if idx := cd.GetSourceIndex(&src.SourceMeta); idx < 0 {
+			cd.Sources = append(cd.Sources, *src)
+		} else {
+			cd.Sources[idx] = *src
+		}
 	}
 
 	return nil
 }
+
+func fetchSource(ctx cpi.Context, src source.Source, path, repo, version string) (*ocm.Source, error) {
```

Comment:
Don't think you need to wrap this again. This function does nothing else but calling another function.